pipelines/generate_epiweekly.R

```diff
@@ -0,0 +1,96 @@
+script_packages <- c(
+  "arrow",
+  "argparser",
+  "dplyr",
+  "forecasttools",
+  "fs",
+  "readr"
+)
+
+## load in packages without messages
+purrr::walk(script_packages, \(pkg) {
+  suppressPackageStartupMessages(
+    library(pkg, character.only = TRUE)
+  )
+})
+
+#' Convert Daily Data to Epiweekly Data
+#'
+#' This function reads daily data from a CSV file, converts it to epiweekly
+#' data, and writes the resulting epiweekly data to a new CSV file.
+#'
+#' @param model_run_dir A string specifying the directory containing the model
+#'  run data.
+#' @param strict A logical value indicating whether to enforce strict inclusion
+#' of only full epiweeks. Default is TRUE.
+#' @param day_of_week An integer specifying the day of the week to use for the
+#' epiweek date. Default is 1 (Monday).
+#'
+#' @return None. The function writes the epiweekly data to a CSV file in the
+#'  specified directory.
+convert_daily_to_epiweekly <- function(
+    model_run_dir, strict = TRUE,
+    day_of_week = 7, dataname = "data", ext = "csv") {
```

Comment:
I think it makes more sense to take the `data_path` as the arg and use utilities from `fs` to extract the parts as necessary.

Reply:
I left `dataname` as an arg, because `model_run_dir` is a common arg across multiple processes (although obvs we could change this in the associated `main` function so feel free to come back at me on that). 

However, I agree than the `ext` doesn't need to be a separate arg!